conformance/utils/suite/experimental_profiles.go

```diff
@@ -0,0 +1,155 @@
+	HTTPConformanceProfileName ConformanceProfileName = "HTTP"
+
+	// TLSConformanceProfileName indicates the name of the conformance profile
+	// which covers TLS stream functionality, such as the TLSRoute API.
+	TLSConformanceProfileName ConformanceProfileName = "TLS"
+
+	// MeshConformanceProfileName indicates the name of the conformance profile
+	// which covers service mesh functionality.
+	MeshConformanceProfileName ConformanceProfileName = "MESH"
+)
+
+// -----------------------------------------------------------------------------
+// Conformance Profiles - Public Vars
+// -----------------------------------------------------------------------------
+
+var (
+	// HTTPConformanceProfile is a ConformanceProfile that covers testing HTTP
+	// related functionality with Gateways.
+	HTTPConformanceProfile = ConformanceProfile{
+		Name: HTTPConformanceProfileName,
+		CoreFeatures: sets.New(
+			SupportGateway,
+			SupportReferenceGrant,
+			SupportHTTPRoute,
+		),
+		ExtendedFeatures: HTTPExtendedFeatures,
+	}
+
+	// TLSConformanceProfile is a ConformanceProfile that covers testing TLS
+	// related functionality with Gateways.
+	TLSConformanceProfile = ConformanceProfile{
+		Name: TLSConformanceProfileName,
+		CoreFeatures: sets.New(
+			SupportGateway,
+			SupportReferenceGrant,
+			SupportTLSRoute,
+		),
+	}
+
+	// MeshConformanceProfile is a ConformanceProfile that covers testing
+	// service mesh related functionality.
+	MeshConformanceProfile = ConformanceProfile{
+		Name: MeshConformanceProfileName,
+		CoreFeatures: sets.New(
+			SupportMesh,
+		),
+	}
+)
+
+// -----------------------------------------------------------------------------
+// Conformance Profiles - Private Profile Mapping Helpers
+// -----------------------------------------------------------------------------
+
+// conformanceProfileMap maps short human-readable names to their respective
+// ConformanceProfiles.
+var conformanceProfileMap = map[ConformanceProfileName]ConformanceProfile{
+	HTTPConformanceProfileName: HTTPConformanceProfile,
+	TLSConformanceProfileName:  TLSConformanceProfile,
+	MeshConformanceProfileName: MeshConformanceProfile,
+}
+
+// getConformanceProfileForName retrieves a known ConformanceProfile by it's simple
+// human readable ConformanceProfileName.
+func getConformanceProfileForName(name ConformanceProfileName) (ConformanceProfile, error) {
+	profile, ok := conformanceProfileMap[name]
+	if !ok {
+		return profile, fmt.Errorf("%s is not a valid conformance profile", name)
+	}
+
+	return profile, nil
+}
+
+// getConformanceProfilesForTest retrieves the ConformanceProfiles a test belongs to
+// given the name of the test.
+//
+// TODO: this is a hack right now using the name of the test itself to determine
+// what profile a test belongs to. If we take this past the
+// Prototyping/Provisional phase we should look into associating profiles more
+// directly with the test (perhaps ON the tests like features).
```

Comment:
I'm not sure why we need this workaround? We already have an exported `ConformanceTests` var, and each item in that list includes a list of `suite.SupportedFeature`. We also have `sets` that include the features in each profile (example: https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/utils/suite/features.go#L48). Relying on a prefix of the test name feels unnecessarily fragile and complex.

If I'm understanding this correctly, it means that each test can fit into exactly one bucket. Instead I think we want tests to be counted for every bucket they are relevant to.

Reply:
I refactored the function, actually, that workaround wasn't really necessary. Please see 6393be5